feat: added loading animations on changing address on checkout

N-Feist opened the pull request:
## Why:

Closes: [[AB#160016](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/160016)]

## Describe your changes

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

N-Feist committed:
feat: added loading animations on changing address on checkout

N-Feist committed:
fix: format fix

csandru-plenty commented on apps/web/components/AddressContainer/AddressContainer.vue:
```diff
@@ -61,7 +64,7 @@ const { disabled = false, type } = defineProps<AddressContainerProps>();
 const isBilling = type === AddressType.Billing;
 const isShipping = type === AddressType.Shipping;
-const { checkoutAddress, hasCheckoutAddress } = useCheckoutAddress(type);
+const { loading: AddressLoading, checkoutAddress, hasCheckoutAddress } = useCheckoutAddress(type);
```
rename  to `addressLoading` please, for consistency.

csandru-plenty commented on apps/web/components/AddressContainer/AddressContainer.vue:
```diff
@@ -49,10 +49,13 @@
       </template>
     </div>
   </div>
+  <div v-else class="flex items-center justify-center relative h-fit min-h-[6rem]">
```
again, for consistency, can you please move this at the top:
```
<div v-if="addressLoading" class="flex items-center justify-center relative h-fit min-h-[6rem]">...</div>

<div v-else data-testid="checkout-address" class="md:px-4 py-6">...</div>
```

N-Feist committed:
fix: renaming variable for consistency

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

maxiroellplenty reviewed (changes requested):
The tasks states: "Put a skeleton in it's place"  like /apps/web/components/blocks/BannerCarousel/BannerSkeleton.vue

The skeleton components should be be named like the main component and in the same folder. Example: 
AddresContainer folder
AddressContainer.vue 
AddressContainerSkeleton.vue

N-Feist committed:
feat: added skeleton to loading animation

N-Feist committed:
chore: improve readability

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
feat: improve code

N-Feist committed:
fix: format fix

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
feat: show skeleton on initial loading

N-Feist committed:
fix: format fix

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

maxiroellplenty commented on apps/web/components/AddressDisplay/AddressDisplaySkeleton.vue:
```diff
@@ -0,0 +1,7 @@
+  <p class="mt-2 h-4 w-36 bg-gray-300 rounded animate-pulse" />
+  <p class="mt-2 h-4 w-36 bg-gray-300 rounded animate-pulse" />
+  <p class="mt-2 h-4 w-36 bg-gray-300 rounded animate-pulse" />
+</template>
```
```suggestion
<template>
  <div class="flex gap-2">
    <span class="h-3 w-20 bg-gray-300 rounded animate-pulse" />
    <span class="h-3 w-20 bg-gray-300 rounded animate-pulse" />
  </div>
  <div class="flex gap-2 mt-2">
    <span class="h-3 w-24 bg-gray-300 rounded animate-pulse" />
    <span class="h-3 w-5 bg-gray-300 rounded animate-pulse" />
  </div>
  <div class="flex gap-2 mt-2">
    <span class="h-3 w-12 bg-gray-300 rounded animate-pulse" />
    <span class="h-3 w-12 bg-gray-300 rounded animate-pulse" />
  </div>
  <p class="mt-2 h-3 w-20 bg-gray-300 rounded animate-pulse" />
</template>
```

more like this, needs to be adjusted in height and margins so that there is little to non shift when the data loads

N-Feist committed:
Update apps/web/components/AddressDisplay/AddressDisplaySkeleton.vue

Co-authored-by: Maximilian Röll <30629157+maxiroellplenty@users.noreply.github.com>

N-Feist committed:
fix: align with real addresscontainer

N-Feist committed:
fix: changed to only the skeleton of addressDisplay

talked to tim about it and it makes more sense for us to only load the AddressDisplaySkeleton here

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
fix: adjust AddressFormSkeleton to the dimensions of the normal AddressForms and show addressformskeleton, when guest checkout

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

N-Feist committed:
fix: lint fix and format fix

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

maxiroellplenty commented on apps/web/components/AddressContainer/AddressContainer.vue:
```diff
@@ -1,58 +1,75 @@
+        <AddressFormSkeleton v-else />
+        <div class="absolute inset-0 z-[999] flex items-center justify-center bg-white/30">
+          <SfLoaderCircular size="2xl" />
+        </div>
```
```suggestion
```

maxiroellplenty replied:
remove the loading spinner, we have the skeletons now ;)

N-Feist replied:
@maxiroellplenty done 👍

N-Feist committed:
fix: removed the loading spinner

maxiroellplenty commented on apps/web/components/AddressContainer/AddressContainer.vue:
```diff
@@ -61,10 +75,21 @@ const { disabled = false, type } = defineProps<AddressContainerProps>();
+      initialFetchingAddress.value = false;
+      stop();
+    }
+  });
```
```suggestion
  watchEffect(() => {
    if (fetchingAddress.value === false) {
      initialFetchingAddress.value = false;
    }
  });
```

maxiroellplenty replied:
https://vuejs.org/guide/essentials/watchers.html#watcheffect
why do we need the stop?

N-Feist replied:
@maxiroellplenty actually, we dont need it ':D i wanted to prevent the watcher to run more than once, but seems like it doens't anyway

N-Feist committed:
Update apps/web/components/AddressContainer/AddressContainer.vue

Co-authored-by: Maximilian Röll <30629157+maxiroellplenty@users.noreply.github.com>

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes

maxiroellplenty committed:
refactor: skeleton display when fetching addresses

maxiroellplenty committed:
chore: format

maxiroellplenty committed:
fix: e2e tests

maxiroellplenty committed:
chore: format

N-Feist committed:
Merge branch 'main' into feat/loading-animation-on-address-changes
